rmf_api_msgs/samples/task_state/multi_dropoff_delivery.json

```diff
@@ -0,0 +1,214 @@
+{
+  "booking": {
+    "id": "delivery_2021:11:08:23:50",
+    "unix_millis_earliest_start_time": 1636388400000,
+    "priority": "none",
+    "automatic": false
+  },
+  "header": {
+    "category": "Multi-Delivery",
+    "detail": [
+      {
+        "category": "Pick Up",
+        "params": {
+          "location": "Kitchen",
+          "items": [
+            {
+              "type": "soda",
+              "quantity": 1
+            },
+            {
+              "type": "water",
+              "quantity": 1
+            }
+          ]
+        }
+      },
+      {
+        "category": "Drop Off",
+        "params": {
+          "location": "room_203",
+          "items": [
+            {
+              "type": "soda",
+              "quantity": 1
+            }
+          ]
+        }
+      },
+      {
+        "category": "Drop Off",
+        "params": {
+          "location": "room_521",
+          "items": [
+            {
+              "type": "water",
+              "quantity": 1
+            }
+          ]
+        }
+      }
+    ],
+    "estimate_millis": 2000000
+  },
+  "completed": [
+    {
+      "phase": {
+        "id": 1,
+        "header": {
+          "category": "Pick Up",
+          "detail": {
+            "location": "Kitchen",
+            "items": [
+              {
+                "type": "soda",
+                "quantity": 1
+              },
+              {
+                "type": "water",
+                "quantity": 1
+              }
+            ]
+          },
+          "estimate_millis": 600000
+        },
+        "final_event_id": 0,
+        "events": [
+          {
+            "id": 0,
+            "status": "completed",
```

Comment:
I feel `events` might be a misleading name, usually I see event as some things that happened in a phase. It's weird for an event to be mutable.

Reply:
I get what you mean, but we kind of beat this horse to death in the group discussions, and `Event` came out as the winner. The other ideas we considered were `Condition`, `Progress`, `Requisite`, and `Checkpoint`, but none of them were very compelling.

In this case, `events` is referring to any "events" pertaining to the phase, whether those events _have_ happened, _might_ happen, or _need_ to happen. I understand why you would naturally assume that "events" would only be past tense, but if you focus on the [enum of the event status](https://github.com/open-rmf/rmf_api_msgs/blob/daa81bc7cf802902a611b2975ae561ba42c5b94c/rmf_api_msgs/schemas/task_state.schema.json#L110) you'll see that an event may be `standby` (meaning it hasn't started yet), `underway` (meaning progress is being made on it), `delayed` (meaning it's taking longer than it was supposed to), `completed` (meaning it was successfully finished) or several other possible statuses. Taking that into account, I think it becomes more clear that this isn't referring only to something that has passed.

The kind of immutable event tracking that you have in mind will be provided by the `log`.

Reply:
It's just minor naming preference, I have no strong objections as well.